### PR TITLE
Serialize CurrencySnapshot using circe

### PR DESCRIPTION
**Pull request by kriscieslak**

### CODE REVIEW

File: modules/shared/src/main/scala/org/tessellation/json/JsonSerializer.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.tessellation.json
+
+import io.circe.jawn.JawnParser
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder}
+
+object JsonSerializer {
+  def serialize[A: Encoder](content: A) =
```
Comment (marcinwadon) [inline modules/shared/src/main/scala/org/tessellation/json/JsonSerializer.scala:8]: Missing types

**Review by marcinwadon — APPROVED** [review]
LGTM, but imo there should be return types

File: modules/shared/src/main/scala/org/tessellation/json/JsonSerializer.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    content.asJson.noSpaces.getBytes("UTF-8")
+
+  def deserialize[A: Decoder](content: Array[Byte]): Either[Throwable, A] =
+    JawnParser(false).decodeByteArray(content)
```
Comment (TheMMaciek) [inline modules/shared/src/main/scala/org/tessellation/json/JsonSerializer.scala:12]: Specifying charset is not necessary?

**Review by TheMMaciek — CHANGES_REQUESTED** [review]
We also need to use this functionality in currency L1 -> currency snapshot processor still uses kryo to deserialize binary.

File: modules/shared/src/main/scala/org/tessellation/json/JsonSerializer.scala (hunk 2 above)
Comment (kriscieslak) [reply modules/shared/src/main/scala/org/tessellation/json/JsonSerializer.scala:12]: There is no parameter to pass charset, and I've checked, that under the hood utf-8 is used.

**Comment by cngo-github** [comment]
@sdavidp for reference

File: modules/shared/src/test/scala/org/tessellation/json/JsonSerializerSuite.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    currencyIncrementalSnapshot[IO](Hash.empty, CurrencySnapshotInfo(SortedMap.empty, SortedMap.empty)).map { signedSnapshot =>
+      val serialized = JsonSerializer.serialize(signedSnapshot)
+      val deserialized = JsonSerializer.deserialize[CurrencySnapshot](serialized)
+      expect.same(true, deserialized.isLeft)
+    }
+  }
+
+  private def currencyIncrementalSnapshot[F[_]: MonadThrow: KryoSerializer](
+    hash: Hash,
+    currencySnapshotInfo: CurrencySnapshotInfo
```
Comment (TheMMaciek) [inline modules/shared/src/test/scala/org/tessellation/json/JsonSerializerSuite.scala:49]: You are using this function twice with the same "empty" value for this parameter. You could remove the parameter altogether or give a default value for it if you think injecting different value will be useful in the future.

File: modules/shared/src/main/scala/org/tessellation/statechannel/StateChannelSnapshotBinary.scala [hunk 4]
```diff
@@ -7,8 +7,9 @@ import org.tessellation.schema.arrayOrder
 import org.tessellation.security.hash.Hash
 
 import derevo.cats.{eqv, order}
-import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
+import derevo.circe.magnolia.decoder
+import derevo.circe.magnolia.encoder
```
Comment (TheMMaciek) [inline modules/shared/src/main/scala/org/tessellation/statechannel/StateChannelSnapshotBinary.scala:12]: No change needed in this file

**Review by TheMMaciek — APPROVED** [review]

**Review by TheMMaciek — APPROVED** [review]

**Pull request merged**